src/script/main.ts

```diff
@@ -1,10 +1,26 @@
+import { stat } from "fs";
+import { isNullOrUndefined } from "util";
+
 export class Main {
     public static viPath;
+    public static viModel;
 
     public static LoadFile(filePath: string) {
-        const currentState = history.state;
+        const currentState = top.history.state;
         const search = "?q=" + filePath;
         if (currentState !== null) {
+            /* As few file models(like LVClass/ PolyVi) donot have # values.
```

Comment:
Technically, the models don't have # values at all. You can just change it to this:
/* As few file types (like LVClass/ PolyVI) do not have # values, so store an empty string instead of storing an undefined value. */

Reply:
Hey, Ayush, can you please copy paste this comment there?
/* As a few file types (like LVClass/ PolyVI) do not have # values, we store an empty string in the session storage instead of storing an undefined value. */

I've made some minor corrections, like file *types* in place of file *models* (that's what I meant in the previous comment when I said that the models don't have # values. I meant that the _models_ don't have # values, their corresponding _views_ have the # values), and other minor things like changing some punctuation and typos. And I made a couple of edits to the previous suggestion as well. Could you please copy-paste it into the code comment?